src/core/execute.c

```diff
@@ -3852,16 +3842,12 @@ static int apply_mount_namespace(
                                        "Bind mounts: %zu, temporary filesystems: %zu, root directory: %s, root image: %s, dynamic user: %s",
                                        n_bind_mounts, context->n_temporary_filesystems, yes_no(root_dir), yes_no(root_image), yes_no(context->dynamic_user));
 
-                        r = -EOPNOTSUPP;
-                } else {
+                        return -EOPNOTSUPP;
```

Comment:
You can use `return log_unit_debug_errno(u, SYNTHETIC_ERRNO(EOPNOTSUPP), ...)`.

Reply:
Ah, indeed.